### PR TITLE
Explicitly set Hex package files

### LINKED ISSUE #265: Add CHANGELOG to package files
Hey @mtrudel, thanks for Bandit!  😄 

Would you be open to adding the CHANGELOG to the list of files included in the hex package?  I personally find it helpful when comparing package versions using the diff tool on hex.pm (specifically when there's a new release).  If you're not opposed, I'll gladly PR the change.  The current list of package files is:

```elixir
["lib", "test", "mix.exs", "README*", "LICENSE*"]
```

Related, I was wondering if it was intentional (or still desirable) to include the test files in the package?  If so, I'm curious to understand the rationale for doing so.

Finally, if you __don't want__ to include the test directory, and __do want__ to include the CHANGELOG, another option would be to remove the `files:` config entirely, and rely on the hex default for `files:`, which is:

```elixir
[
  "lib",
  "priv",
  ".formatter.exs",
  "mix.exs",
  "README*",
  "readme*",
  "LICENSE*",
  "license*",
  "CHANGELOG*",
  "changelog*",
  "src",
  "c_src",
  "Makefile*"
]
```

See https://hexdocs.pm/hex/Mix.Tasks.Hex.Build.html#module-package-configuration

All of this also applies to [thousand_island](https://github.com/mtrudel/thousand_island), where I'd be happy to PR the change as well if you'd like.  Thanks for your consideration!

### PR DESCRIPTION
**Pull request by wojtekmach**
Closes #265

Sanity check:

    % mix hex.build --unpack -o pkg
    % iex
    Erlang/OTP 26 [erts-14.2.1] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]

    Interactive Elixir (1.17.0-dev) - press Ctrl+C to exit (type h() ENTER for help)
    iex> Mix.install([:req, {:bandit, path: "pkg"}])
    iex> {:ok, _} = Bandit.start_link(scheme: :http, plug: fn conn, _ -> Plug.Conn.send_resp(conn, 200, "hi") end)

    14:12:40.784 [info] Running #Function<41.105768164/2 in :erl_eval.expr/6> with Bandit 1.1.2 at 0.0.0.0:4000 (http)
    iex> Req.get!("http://localhost:4000")
    %Req.Response{
      status: 200,
      headers: %{
        "cache-control" => ["max-age=0, private, must-revalidate"],
        "date" => ["Fri, 29 Dec 2023 13:12:40 GMT"],
        "vary" => ["accept-encoding"]
      },
      body: "hi",
      trailers: %{},
      private: %{}
    }

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -16,7 +16,8 @@ defmodule Bandit.MixProject do
       package: [
         maintainers: ["Mat Trudel"],
         licenses: ["MIT"],
-        links: %{"GitHub" => "https://github.com/mtrudel/bandit"}
+        links: %{"GitHub" => "https://github.com/mtrudel/bandit"},
+        files: ["lib", "mix.exs", "README*", "LICENSE*", "CHANGELOG*"]
```
Comment (wojtekmach) [inline mix.exs:20]: This also might be a good idea:

```suggestion
        files: ["lib", "mix.exs", "*.md"]
```

```
% mix hex.build
Building bandit 1.1.2
  Dependencies:
    thousand_island ~> 1.0 (app: thousand_island)
    plug ~> 1.14 (app: plug)
    websock ~> 0.5 (app: websock)
    hpax ~> 0.1.1 (app: hpax)
    telemetry ~> 0.4 or ~> 1.0 (app: telemetry)
  App: bandit
  Name: bandit
  Files:
    lib
    (...)
    mix.exs
    CHANGELOG.md
    CODE_OF_CONDUCT.md
    README.md
    SECURITY.md
```

**Comment by mtrudel** [comment]
I'll take it from the horse's mouth that this is the right way to do it! Thanks @wojtekmach ; I'll be copying this into my other projects as well.

**Comment by mtrudel** [comment]
Oh wait - what about the stuff in `assets/`; that'll be required to build docs locally, no?

**Comment by wojtekmach** [comment]
assets aren’t used in Hex package. They are present in your local checkout when you run mix docs so it’s all good. :)

**Comment by mtrudel** [comment]
Do I need to?

**Comment by wojtekmach** [comment]
Nope, it's fine as is. Just letting you know there's another way, created basically for this scenario, to solve the problem. :)

**Comment by mtrudel** [comment]
Got it. I'm going to be passing over my other projects and applying this as needed, so it's great to know.

Thanks!

**Pull request merged**